src/plugin.js

```diff
@@ -0,0 +1,15 @@
+import { Inertia } from '@inertiajs/inertia'
+
+import Link from './link'
+import Remember from './remember'
+
+export default {
+  install(Vue) {
+    Object.defineProperty(Vue.prototype, '$inertia', { get: () => Inertia })
+    Object.defineProperty(Vue.prototype, '$page', { get: () => app.props })
```

Comment:
Still an issue with setting the `$page` here.

This wasn't a problem when having `InertiaApp` & `InertiaPlugin` in one.

Reply:
```suggestion
    Object.defineProperty(Vue.prototype, '$page', { get: () => Inertia.page.props })
```

Would this work? This setup feels a bit dirty to me though, since we're pulling state from `Inertia` instead of `InertiaApp`.

Alternatively, we could look into a provide/inject setup.

Reply:
`page` doesn't exist on the `Inertia` instance, so this wouldn't work.

I'll look into  the provide/inject setup you mentioned.

Maybe @reinink has some other ideas too? 🤔